components/constellation/constellation.rs

```diff
@@ -1644,6 +1649,9 @@ where
         let is_private = false;
         let is_visible = true;
 
+        // Register this new top-level browsing context id as a browser and set
+        // its focused browsing context to be itself.
+        self.browser_ids.insert(top_level_browsing_context_id, browsing_context_id);
         if self.focused_browsing_context_id.is_none() {
```

Comment:
This can be removed.

Reply:
Because of the suggestion above to always do an insert when we get a focus message? I think it's cleaner if we insert only during a browser creation and subsequently only do updates in `handle_focus_msg`, i.e.
```rust
self.browsers.get_mut(&top_level_browsing_context_id)
    .focused_browsing_context_id = browsing_context_id;
```
or something similar.

Reply:
Yes this sounds right, I was referring to setting `focused_browsing_context_id`